refactor(cli): update keypair generate command (#1621)
nervosnetwork/axon: core/cli/README.md

# Axon CLI

Axon CLI is a command line tool for Axon framework. It is included in the Axon binary with the command of `cargo build --release`.

## Usage

### Generate Keypair

```bash

> ./target/release/axon generate-keypair -n 1

Reviewer: ```suggestion
mkdir -p ./devtools/chain/data
./target/release/axon generate-keypair -n 1 --path ./devtools/chain/data
```

fix `Error writing private key: No such file or directory (os error 2)``

Author: The path has the default value `/free-space/` don't need to `mkdir`.

Reviewer: > fix `Error writing private key: No such file or directory (os error 2)``

I tried the cmd `./target/release/axon generate-keypair -n 1` on a fresh machine, and it threw the above error.

Author: Because the folder does not have `free-space` direction.

```

Example Output:

```bash
$ cargo run -- -n 1
{
  "keypairs": [
    {
      "index": 0,
      "net_private_key": "0x0000000000000000000000000000000000000000000000000000000000000000",
      "public_key": "0x020044def250d1d27c0f7c0eeee6c8b964756d5ad7936fcefaf3f086245ddb9c9c",
      "address": "0x83fec4b12b26a70195ccabb67ce0cd692856eaf1",
      "peer_id": "QmbZEzvonMiPiioRpYeVWxngjN42FHC3EHXjeo2C7o2NDZ",
      "bls_private_key": "0x1111111111111111111111111111111111111111111111111111111111111111",
      "bls_public_key": "0x81ca54f01e3b433c209ae1d165e6362669e13170f89712dbc52c3572901d025ecc206a43c4b25f58388189f74587a7d"
    }
  ]
}
```

The arguments are described in the following table:

| name | short | default value |
|--|--|--|
| number | n | 4 |
| private-key-path | p | current_path/free-space/ |

### Update Config and Spec File

> Firstly, update the `validator_list` in [chain-spec](../chain/specs/single_node/chain-spec.toml) file. Replace the `bls_pub_key`, `pub_key` and `address` with the value generated by the keypair generator.For example:
> 

Reviewer: ```suggestion
First, update the `validator_list` in the [chain-spec](../../devtools/chain/specs/single_node/chain-spec.toml) file. Replace the `bls_pub_key`, `pub_key` and `address` with the value generated by the keypair generator, for example:

```

```toml
[[params.verifier_list]]
bls_pub_key = "0x81ca54f01e3b433c209ae1d165e6362669e13170f89712dbc52c3572901d025ecc206a43c4b25f58388189f74587a7d"
pub_key = "0x020044def250d1d27c0f7c0eeee6c8b964756d5ad7936fcefaf3f086245ddb9c9c"
address = "0x83fec4b12b26a70195ccabb67ce0cd692856eaf1"
propose_weight = 1
vote_weight = 1
```

> Then, update the `bootstraps` in [config](../chain/config.toml) file. Fill the generated `peer_id` to the `multi_address` For example:
> 

Reviewer: ```suggestion
Then, update the `bootstraps` in the [config](../../devtools/chain/config.toml) file. Fill the generated `peer_id` to the `multi_address` For example:

```

```toml
[[network.bootstraps]]
multi_address = "/ip4/127.0.0.1/tcp/8001/p2p/QmbZEzvonMiPiioRpYeVWxngjN42FHC3EHXjeo2C7o2NDZ"
```

> Finally, set the `net_privkey_file` and `bls_privkey_file` with the generated private key file path in [config](../chain/config.toml) file.
> 

Reviewer: ```suggestion

Finally, set the `net_privkey_file` and `bls_privkey_file` with the generated private key file path in [config](../../devtools/chain/config.toml) file.

```
